### PR TITLE
Datacube initial commit

### PR DESCRIPTION
**Pull request by lonesword**
commcell.datacube.datasources would now print all datasources in a datacube instance.

### CODE REVIEW

**Comment by lonesword** [comment]
Please review

File: cvpysdk/commcell.py [hunk 1]
```diff
@@ -164,6 +165,7 @@ def __init__(self, webconsole_hostname, commcell_username, commcell_password=Non
             UserGroups,
             WorkFlow,
             ClientGroups,
+            Datacube
```
Comment (GoelJatin) [inline cvpysdk/commcell.py]: Missing `,`
Please add it

File: cvpysdk/datacube.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        self._commcell_object = commcell_object
+
+        # APIs
+        self._ANALYTICS_ENGINES = self._commcell_object._services['GET_ANALYTICS_ENGINES']
+        self._ALL_DATASOURCES = self._commcell_object._services['GET_ALL_DATASOURCES']
+
+        # API results
+        self.analytics_engines = self._get_analytics_engines()
+
+        datacube_sdk_classes = [
```
Comment (GoelJatin) [inline cvpysdk/datacube.py]: Will other classes come in this list later.
If not, then don't use this part, and queues

File: cvpysdk/datacube.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        self.datasources = datacube_sdk_dict[Datasources]
+
+    def __str__(self):
+        """Representation string consisting of all datacube of the commcell.
+
+            Returns:
+                str - string of all the clients associated with the commcell
+
+            TODO: Do this
+        """
```
Comment (GoelJatin) [inline cvpysdk/datacube.py]: We'll merge the code, once this part is also done

File: cvpysdk/datacube.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                         "listOfCIServer": [] //array of analytics engines
+                    }
+
+            Raises:
+                SDKException:
+                    if response is empty
+
+                    if response is not success
+        """
+        flag, response = self._commcell_object._cvpysdk_object.make_request\
```
Comment (GoelJatin) [inline cvpysdk/datacube.py]: why split into multiple lines
please correct this

File: cvpysdk/sedstype.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
+
+class SEDSType(object):
```
Comment (GoelJatin) [inline cvpysdk/sedstype.py]: Will this be used across multiple files, or in a single file.
If it'll used in a single file, then please add this dictionary to that file only.

Also, for this operation, I don't think a class is required.
You can directly create the dict here, and take a copy of it in your class

File: cvpysdk/datasource.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+    def _get_cached_datasource(self, datasource_name):
+        for datasource in self._datasources:
+            if datasource['datasourceName'] == datasource_name:
+                return datasource
+
+        return None
+
+    def get(self, datasource_name):
+        """
+        Returns a datasource from datacube that matches the supplied datasource_name
```
Comment (GoelJatin) [inline cvpysdk/datasource.py]: Please follow the same doc string conventions, as in other methods and classes

File: cvpysdk/datacube.py (hunk 2 above)
Comment (klogan12) [inline cvpysdk/datacube.py]: Data cube might not be available on all commcell , can we create getter for  datacube in commcell.py. and that  would return datacube object.

Comment (lonesword) [inline cvpysdk/datacube.py]: Yes, other classes would come in this datacube_sdk_classes later. And yes, I'll write a getter for datacube

File: cvpysdk/datacube.py (hunk 4 above)
Comment (lonesword) [inline cvpysdk/datacube.py]: Linter made me do this, to conform to PEP-8 (with line-length as 99)

File: cvpysdk/sedstype.py (hunk 5 above)
Comment (lonesword) [inline cvpysdk/sedstype.py]: This will be used in multiple files, and hence the separate class in a separate file

File: cvpysdk/commcell.py (hunk 1 above)
Comment (lonesword) [inline cvpysdk/commcell.py]: oops, merge conflict. Will fix. I wonder how it did not throw an error when I tested

**Comment by lonesword** [comment]
Made the CR changes. Please review

**Pull request merged**